src/status_im/contexts/chat/messenger/composer/view.cljs

```diff
@@ -1,183 +1,56 @@
-           :input-ref    (:input-ref props)}]]
-        [reanimated/touchable-opacity
-         {:active-opacity      1
-          :on-press            (fn []
-                                 (when-let [ref @(:input-ref props)]
-                                   (.focus ^js ref)))
-          :style               (style/input-container (:height animations) max-height)
-          :accessibility-label :message-input-container}
-         [rn/selectable-text-input
-          {:ref        #(reset! (:selectable-input-ref props) %)
-           :menu-items @(:menu-items state)
-           :style      (style/input-view state)}
-          [rn/text-input
-           {:ref                      #(reset! (:input-ref props) %)
-            :default-value            @(:text-value state)
-            :on-focus                 #(handler/focus props state animations dimensions)
-            :on-blur                  #(handler/blur state animations dimensions)
-            :on-content-size-change   #(handler/content-size-change %
-                                                                    state
-                                                                    animations
-                                                                    dimensions
-                                                                    (or keyboard-shown
-                                                                        (:edit subscriptions)))
-            :on-scroll                #(handler/scroll % props state animations dimensions)
-            :on-change-text           #(handler/change-text % props state)
-            :on-selection-change      #(handler/selection-change % props state)
-            :on-selection             #(selection/on-selection % props state)
-            :keyboard-appearance      theme
-            :max-font-size-multiplier 1
-            :multiline                true
-            :placeholder              (i18n/label :t/type-something)
-            :placeholder-text-color   (colors/theme-colors colors/neutral-40 colors/neutral-50 theme)
-            :style                    (style/input-text props
-                                                        state
-                                                        {:max-height max-height
-                                                         :theme      theme})
-            :max-length               constants/max-text-size
-            :accessibility-label      :chat-message-input}]]]
-        [:<>
-         [gradients/view props state animations show-bottom-gradient?]
-         [link-preview/view]
-         [images/images-list]]
-        [:f> actions/view props state animations window-height subscriptions]]]]]))
+(defn input
+  [_ _]
+  (let [default-value (:input-text (rf/sub [:chats/current-chat-input]))]
+    (fn [set-ref theme]
+      [rn/text-input
+       {:ref                      set-ref
+        :on-change-text           handler/change-text
```

Comment:
Now that we are trying to avoid Reagent's form-2 components we could use a `use-state` initialized with the subscription's value instead.

Btw, shouldn't we have a subscription to directly extract the text instead of always extracting from subscription's return value?

Reply:
yes, this is a correct observation, but because input-field is broken or idk, it works wrong, we don't want to update it with default-value, so I locked it in the init function, I didn't find a good solution with the use-state,  the problem is that subscription is reactive, it changes each time user updates input, but we want to use it only once on component init, we don't want to react each time user updated input